packages/googlecloud/functions/getanswer/inquirer.py

```diff
@@ -63,11 +66,101 @@ def get_response_from_query(db, query, k=4):
+
+    Note: If the available information from the transcripts is insufficient to accurately answer the question or recreate the dialogue, please respond with 'Insufficient information available.' If the question extends beyond the scope of information contained in the transcripts, state 'I don't know.'
+    """,
+    )
+    chain_llm = LLMChain(llm=llm, prompt=prompt)
+    responses_llm = chain_llm.run(question=query, docs=docs_page_content)
+
+    generated_responses = responses_llm.split("\n\n")
+
+    generated_titles = [doc.metadata["title"] for doc in docs]
+    publish_dates = [doc.metadata["publish_date"].strftime("%Y-%m-%d") for doc in docs]
+
+    final_response = ""
+    for response, title, publish_date in zip(
+        generated_responses, generated_titles, publish_dates
+    ):
+        final_response += (
+            remove_numbering_prefix(response)
+            + f"\n\nSourced from Youtube: {title} (Published on: {publish_date})\n\n"
+        )
+
+    return final_response
+
+
+def get_general_summary_response_from_query(db, query, k=4):
+    logger.info("Performing general summary query...")
+    llm = ChatOpenAI(model_name="gpt-3.5-turbo-0613")
+
+    docs = db.similarity_search(query, k=k)
+
+    docs_page_content = " ".join([d.page_content for d in docs])
+    prompt = PromptTemplate(
+        input_variables=["question", "docs"],
+        template="""
+    As an AI assistant, you are to recreate the actual dialogue that occurred between city council members and external stakeholders, based on the transcripts from New Orleans City Council meetings provided in "{docs}".
+
+    In response to the question "{question}", your output should mimic the structure of a real conversation, which often involves more than two exchanges between the parties. As such, please generate as many pairs of statements and responses as necessary to completely answer the query. 
+
+    For each statement and response, provide a summary. The response should take the following format:
+
+    1. Summary of Statement from City Council Member.
+    2. Summary of Response or Statement external stakeholder
+    3. Continue this pattern, including additional statements and responses from both parties as necessary to provide a comprehensive answer.
+    4. No direct quotes should be included, only summaries.
+
+    Note: If the available information from the transcripts is insufficient to accurately answer the question or recreate the dialogue, please respond with 'Insufficient information available.' If the question extends beyond the scope of information contained in the transcripts, state 'I don't know.'
+    """,
+    )
+    chain_llm = LLMChain(llm=llm, prompt=prompt)
+    responses_llm = chain_llm.run(question=query, docs=docs_page_content)
+
+    return responses_llm
+
+
+def route_question(db, query, k=4):
```

Comment:
I don't love this approach tbh. We'll need to generate a bunch of categories. And it won't  handle homeless issues that involve crime well. Maybe this is the best way to do it quick and dirty though?

Why did you decide on this approach vs doing one that's based on length like my cuz suggested?

Reply:
1) This approach is distinct from the suggestion from cuz. Homelesss is just an example that I used to get your feedback. The idea is that the model can provide better responses to queries based on the prompt. From my understanding, the k parameter is used in conjunction with the prompt to find the documents most similar to the query, therefore, a general prompt, like return a dialogue based on this query, will provide worse results than when routed to a prompt that is tailored toward a specific topic. For this to work, yea, we'd have to generate a bunch of categories. 

2) The approach your cuz suggested is addressed in the function returned by the else clause below. It can for sure be refined to just provide a few sentences, rather than a dialogical summary. 
```
def get_general_summary_response_from_query(db, query, k=4):
    logger.info("Performing general summary query...")
    llm = ChatOpenAI(model_name="gpt-3.5-turbo-0613")

    docs = db.similarity_search(query, k=k)

    docs_page_content = " ".join([d.page_content for d in docs])
    prompt = PromptTemplate(
        input_variables=["question", "docs"],
        template="""
    As an AI assistant, you are to recreate the actual dialogue that occurred between city council members and external stakeholders, based on the transcripts from New Orleans City Council meetings provided in "{docs}".

    In response to the question "{question}", your output should mimic the structure of a real conversation, which often involves more than two exchanges between the parties. As such, please generate as many pairs of statements and responses as necessary to completely answer the query. 

    For each statement and response, provide a summary. The response should take the following format:

    1. Summary of Statement from City Council Member.
    2. Summary of Response or Statement external stakeholder
    3. Continue this pattern, including additional statements and responses from both parties as necessary to provide a comprehensive answer.
    4. No direct quotes should be included, only summaries.

    Note: If the available information from the transcripts is insufficient to accurately answer the question or recreate the dialogue, please respond with 'Insufficient information available.' If the question extends beyond the scope of information contained in the transcripts, state 'I don't know.'
    """,
    )
    chain_llm = LLMChain(llm=llm, prompt=prompt)
    responses_llm = chain_llm.run(question=query, docs=docs_page_content)

    return responses_llm
```

Reply:
1. I think you're right for the first iteration of the project. I'll work on revising the prompt to generate the most optimal response irrespective of the query. 
2. Adjusted the general model to be more aligned with cuzzos suggestion